libpod/runtime_img.go

```diff
@@ -171,21 +254,36 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 	}
 	defer policyContext.Destroy()
 
-	copyOptions := common.GetCopyOptions(reportWriter, signaturePolicyPath, nil, nil, common.SigningOptions{})
+	copyOptions := common.GetCopyOptions(reportWriter, signaturePolicyPath, nil, &options.DockerRegistryOptions, options.SigningOptions)
+
 	for _, image := range images {
+		fmt.Printf("Trying to pull %s...", image)
 		reference := image
-		if srcRef.DockerReference() != nil {
-			reference = srcRef.DockerReference().String()
+		var pullRef types.ImageReference
+		if fromRegistry {
+			// need to get ImageReference
+			pullRef, err = alltransports.ParseImageName(image)
```

Comment:
This shouldn’t be needed, see the comments around the call to `getRegistriesToTry`.

Also, it would be nice to if the `fromRegistry`/`srcRef` paths shared even more code, perhaps by first building a “task list” data structure (a list of destinations and sources), and then just executing it.